csi/src/node/nvmfutil.rs

```diff
@@ -127,10 +129,25 @@ fn nvmeadm_attach_disk(
 
 pub fn nvmeadm_detach_disk(nqn: &str) -> Result<(), Error> {
     match nvmeadm::nvmf_discovery::disconnect(&nqn) {
-        Ok(_) => {
-            trace!("nvmf disconnected {}", nqn);
-            Ok(())
-        }
+        Ok(n) => match n {
+            0 => {
+                debug!("nvmf disconnect {} FAILED, no device found.", nqn);
+                Err(Error::from(CSIError::NotFound {
+                    value: format!("nvmf device not found for {}", nqn),
+                }))
+            }
+            1 => {
+                trace!("nvmf disconnected {}", nqn);
+                Ok(())
+            }
+            _ => {
+                debug!(
+                    "Warning: nvmf disconnect {} disconnected {} devices.",
+                    nqn, n
+                );
```

Comment:
it's a debug message beginning with "warning:". Perhaps warn! should be used instead (and `warning:` removed)?

Reply:
https://github.com/openebs/Mayastor/pull/307/commits/0cc0a067896409e67197f6a1035499c0c68e4558